controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -274,6 +291,151 @@ func (w *Workload) UpdateKubernetesVersionInKubeadmConfigMap(ctx context.Context
 	return nil
 }
 
+// UpdateCoreDNS updates the kubeadm configmap, coredns corefile and coredns
+// deployment.
+func (w *Workload) UpdateCoreDNS(ctx context.Context, dns *kubeadmv1.DNS) error {
+	if err := w.UpdateCoreDNSImageInfoInKubeadmConfigMap(ctx, dns); err != nil {
+		return err
+	}
+
+	if err := w.UpdateCoreDNSCorefile(ctx, dns); err != nil {
+		return err
+	}
+
+	if err := w.UpdateCoreDNSDeployment(ctx, dns); err != nil {
+		return err
+	}
+	return nil
+}
+
+// UpdateCoreDNSImageInfoInKubeadmConfigMap updates the kubernetes version in the kubeadm config map.
+func (w *Workload) UpdateCoreDNSImageInfoInKubeadmConfigMap(ctx context.Context, dns *kubeadmv1.DNS) error {
+	if dns.Type != kubeadmv1.CoreDNS {
```

Comment:
Given that this is a default, we might want to tweak the check a little: `dns.Type != "" && dns.Type != kubeadmv1.CoreDNS`

Reply:
I think I'm going to move this check out from here and into the `validateCoreDNSUpgrade`